kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeOutputCell.js

```diff
@@ -239,57 +230,56 @@ define([
-                        } else {
-                            $headController.show();
-                            this.headerShown = true;
-                        }
-                    }.bind(this));
+                    .click(
+                        function () {
+                            if (this.headerShown) {
+                                $headController.hide();
+                                this.headerShown = false;
+                            } else {
+                                $headController.show();
+                                this.headerShown = true;
+                            }
+                        }.bind(this)
+                    );
                 this.$body = $('<div class="kb-cell-output-content">')
                     .append($headController)
                     .append($headerBtn);
                 this.$elem.append(this.$body);
             }
 
-            return this.renderBody()
-                .then(function() {
+            return this.renderBody().then(
+                function () {
                     this.isRendered = true;
-                }.bind(this));
+                }.bind(this)
+            );
         },
 
-        renderBody: function() {
+        renderBody: function () {
             var self = this;
-            return new Promise(function(resolve) {
+            return new Promise(function (resolve) {
                 var widget = self.options.widget,
                     widgetData = self.options.data;
                 if (widget === 'kbaseDefaultNarrativeOutput') {
                     widgetData = { data: self.options.data };
                 }
                 widgetData.upas = self.options.upas;
 
-                require([widget],
-                    function (W) {
-                        if (self.$widgetBody) {
-                            self.$widgetBody.remove();
-                        }
-                        self.$widgetBody = $('<div>');
-                        self.$body.append(self.$widgetBody);
-                        self.$outWidget = new W(self.$widgetBody, widgetData);
-                        resolve();
-                    },
-                    // TODO: No, should reject the promise, or handle here and resolve,
-                    // otherwise on error the promise will dangle.
-                    function (err) {
-                        return self.renderError(err);
+                require([widget], function (W) {
+                    if (self.$widgetBody) {
+                        self.$widgetBody.remove();
                     }
-                );
+                    self.$widgetBody = $('<div>');
+                    self.$body.append(self.$widgetBody);
+                    self.$outWidget = new W(self.$widgetBody, widgetData);
+                    resolve();
+                }, // otherwise on error the promise will dangle. // TODO: No, should reject the promise, or handle here and resolve,
```

Comment:
The AST has been changed here. I've reformatted these comments in a later commit as they don't look good like this.